iRecruit-ui/src/components/Home/Home.jsx

```diff
@@ -1,21 +1,78 @@
-import React from "react";
+// import React from "react";
+// import "./Home.css";
+
+// export default function Home({
+//   posts,
+// }) {
+//   return (
+//     <div className="home">
+//       <div className="posts-container">
+//         {posts.map((job) => (
+//           <div className="post" key={job.id}>
+//             <h2>{job.title}</h2>
+//             <h3>{job.company}</h3>
+//             <p>Date Added: {job.dateAdded}</p>
+//           </div>
+//         ))}
+//       </div>
+//     </div>
+//   );
+// }
+
+import React, { useState, useEffect } from "react";
 import "./Home.css";
 
-export default function Home({
-  posts,
-}) {
+export default function Home({ posts, handleChange, handleSubmit, categoryFilter }) {
+  const [searchValue, setSearchValue] = useState("");
+  const [filteredPosts, setFilteredPosts] = useState([]);
+
+  useEffect(() => {
+    filterPosts(posts, searchValue, categoryFilter);
+  }, [posts, searchValue, categoryFilter]);
+
+  const handleSearch = (event) => {
+    const searchInput = event.target.value;
+    setSearchValue(searchInput);
+  };
+
+  const filterPosts = (posts, searchInput, category) => {
+    const filteredPosts = posts.filter((post) => {
+      const { title } = post;
+      const searchQuery = searchInput.toLowerCase();
+
+      if (category === "jobs") {
+        return title && title.toLowerCase().includes(searchQuery);
+      } else {
+        return title && title.toLowerCase().includes(searchQuery);
+      }
```

Comment:
??? what is this doing

Reply:
I see, I basically repeated the same thing, I was trying to say that if the current category is currently on jobs, when the search bar takes a query/input, filter that "current category" by the title that includes the query as a subsequent subset. And then I wanted to say, if we are not on default "jobs", filter by the category we are on.

But I can see now that I could just take out the conditional and leave just the return statement, since category will always exist and we are performing the same function regardless.